src/silo/query_engine/query_plan.cpp

```diff
@@ -32,15 +32,42 @@ arrow::Status QueryPlan::executeAndWriteImpl(
    SPDLOG_TRACE("{}", arrow_plan->ToString());
    arrow_plan->StartProducing();
    SPDLOG_TRACE("Plan started producing, will now read the resulting batches.");
+
+   // Ensure plan is stopped on any exit path (timeout/error/exception).
+   struct PlanStopGuard {
+      std::shared_ptr<arrow::acero::ExecPlan> plan;
+      ~PlanStopGuard() {
+         constexpr double kGraceShutdownSeconds = 5.0;  // avoid hanging on teardown
```

Comment:
Why is there a k in the name?
```suggestion
         constexpr double graceShutdownSeconds = 5.0;  // avoid hanging on teardown
```